From: Linux Kernel Mailing List <linux-kernel@vger.kernel.org>
To: git-commits-head@vger.kernel.org
Subject: asm-{alpha,h8300,um,v850,xtensa}/param.h: unbreak HZ for userspace
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
X-Git-Commit: b7cffc1f29c1bc729bc50c863c87f93f9b70994b
X-Git-Parent: 772279c5f1dceb58d451dca94b557fd89b1ce890
X-Greylist: Sender IP whitelisted, not delayed by milter-greylist-4.0rc1 (hera.kernel.org [127.0.0.1]); Thu, 15 May 2008 02:59:20 +0000 (UTC)
Sender: git-commits-head-owner@vger.kernel.org
Precedence: bulk
List-ID: <git-commits-head.vger.kernel.org>
X-Mailing-List:	git-commits-head@vger.kernel.org

Gitweb:     http://git.kernel.org/git/?p=linux/kernel/git/torvalds/linux-2.6.git;a=commit;h=b7cffc1f29c1bc729bc50c863c87f93f9b70994b
Commit:     b7cffc1f29c1bc729bc50c863c87f93f9b70994b
Parent:     772279c5f1dceb58d451dca94b557fd89b1ce890
Author:     Mike Frysinger <vapier@gentoo.org>
AuthorDate: Wed May 14 16:05:42 2008 -0700
Committer:  Linus Torvalds <torvalds@linux-foundation.org>
CommitDate: Wed May 14 19:11:14 2008 -0700

    asm-{alpha,h8300,um,v850,xtensa}/param.h: unbreak HZ for userspace
    
    I noticed this because alpha was broken due to the recent commit commit
    bdc807871d58285737d50dc6163d0feb72cb0dc2 ("avoid overflows in
    kernel/time.c").  Most arches do something like this in their
    asm/param.h:
    
    #ifdef __KERNEL__
    # define HZ CONFIG_HZ
    #else
    # define HZ 100
    #endif
    
    A few arches though (namely alpha/h8300/um/v850/xtensa) either do no set
    HZ at all for !__KERNEL__, or they set it wrongly.  This should bring all
    arches in line by setting up HZ for userspace.
    
    Without this currently perl 5.10 doesn't build on alpha:
    
    perl.c: In function 'perl_construct':
    perl.c:388: error: 'CONFIG_HZ' undeclared (first use in this function)
    -> http://buildd.debian.org/fetch.cgi?pkg=perl;ver=5.10.0-10;arch=alpha;stamp=1210252894
    
    Signed-off-by: Mike Frysinger <vapier@gentoo.org>
    Cc: Richard Henderson <rth@twiddle.net>
    Cc: Ivan Kokshaysky <ink@jurassic.park.msu.ru>
    Cc: Yoshinori Sato <ysato@users.sourceforge.jp>
    Cc: Jeff Dike <jdike@addtoit.com>
    Cc: Chris Zankel <chris@zankel.net>
    Cc: maximilian attems <max@stro.at>
    Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
    [ HZ on alpha is 1024 for historical reasons.  - Linus ]
    Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>
---
 include/asm-alpha/param.h  |    4 ++++
 include/asm-h8300/param.h  |    8 +++-----
 include/asm-um/param.h     |    2 ++
 include/asm-v850/param.h   |    2 ++
 include/asm-xtensa/param.h |    2 ++
 5 files changed, 13 insertions(+), 5 deletions(-)

diff --git a/include/asm-alpha/param.h b/include/asm-alpha/param.h
index 0982f1d..e691ecf 100644
--- a/include/asm-alpha/param.h
+++ b/include/asm-alpha/param.h
@@ -5,8 +5,12 @@
    hardware ignores reprogramming.  We also need userland buy-in to the 
    change in HZ, since this is visible in the wait4 resources etc.  */
 
+#ifdef __KERNEL__
 #define HZ		CONFIG_HZ
 #define USER_HZ		HZ
+#else
+#define HZ		1024
+#endif
 
 #define EXEC_PAGESIZE	8192
 
diff --git a/include/asm-h8300/param.h b/include/asm-h8300/param.h
index 04f64f1..1c72fb8 100644
--- a/include/asm-h8300/param.h
+++ b/include/asm-h8300/param.h
@@ -1,14 +1,12 @@
 #ifndef _H8300_PARAM_H
 #define _H8300_PARAM_H
 
-
-#ifndef HZ
-#define HZ CONFIG_HZ
-#endif
-
 #ifdef __KERNEL__
+#define HZ		CONFIG_HZ
 #define	USER_HZ		HZ
 #define	CLOCKS_PER_SEC	(USER_HZ)
+#else
+#define HZ		100
 #endif
 
 #define EXEC_PAGESIZE	4096
diff --git a/include/asm-um/param.h b/include/asm-um/param.h
index 4cd4a22..e44f4e6 100644
--- a/include/asm-um/param.h
+++ b/include/asm-um/param.h
@@ -13,6 +13,8 @@
 #define HZ CONFIG_HZ
 #define USER_HZ	100	   /* .. some user interfaces are in "ticks" */
 #define CLOCKS_PER_SEC (USER_HZ)  /* frequency at which times() counts */
+#else
+#define HZ 100
 #endif
 
 #endif
diff --git a/include/asm-v850/param.h b/include/asm-v850/param.h
index 2818326..4391f5f 100644
--- a/include/asm-v850/param.h
+++ b/include/asm-v850/param.h
@@ -26,6 +26,8 @@
 # define HZ		CONFIG_HZ
 # define USER_HZ	100
 # define CLOCKS_PER_SEC	USER_HZ
+#else
+# define HZ		100
 #endif
 
 #endif /* __V850_PARAM_H__ */
diff --git a/include/asm-xtensa/param.h b/include/asm-xtensa/param.h
index 82ad34d..ba03d5a 100644
--- a/include/asm-xtensa/param.h
+++ b/include/asm-xtensa/param.h
@@ -15,6 +15,8 @@
 # define HZ		CONFIG_HZ	/* internal timer frequency */
 # define USER_HZ	100		/* for user interfaces in "ticks" */
 # define CLOCKS_PER_SEC (USER_HZ)	/* frequnzy at which times() counts */
+#else
+# define HZ		100
 #endif
 
 #define EXEC_PAGESIZE	4096
